### PR TITLE
[GPU] OneDNN v3.5 update and fix for OOB memory access

### PR DESCRIPTION
**Pull request by isanghao**
### Details:
 - Update OneDNN to v3.5
 - Fix OOB memory access with fake alignment
    - Do not fake-align if input-memory is not allocated properly (it impacts test cases where FC gets input from data node)
    - This fix is applied only to some hardware because of performance concern

### Tickets:
 - 141215
 - 138012

### CODE REVIEW

**Review by p-durandin — APPROVED** [review]

**Review by yeonbok — APPROVED** [review]

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp [hunk 1]
```diff
@@ -459,6 +459,28 @@ void primitive_inst::update_shape() {
...
+kernel_impl_params primitive_inst::get_fake_aligned_params_if_possible(kernel_impl_params const& orig_impl_param) {
+    auto updated_params = _node->type()->get_fake_aligned_params(orig_impl_param);
+
+    const auto &dev_info = get_node().get_program().get_engine().get_device_info();
+
+    // The target HW of this patch is limited because of performance concern
+    if (dev_info.supports_immad && dev_info.dev_type == device_type::integrated_gpu) {
+        // Check whether the input node has enough space for output data. Otherwise, fake alignment is not possible due to page fault
+        // i.e. predecessor node was supposed be increased already
+        if (get_node().is_type<fully_connected>() && dependencies().size() > 0 && dep_memory(0).get_layout().is_static()
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/primitive_inst.cpp:471]: Can it be a part of fully_connected_inst::get_fake_aligned_params ? There are few checks on fake alignment applicability, so why do we need to move these checks outside?

Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp:471]: Currently it is declared as static function. I think we can apply this within the existing get_fake_aligned_params if we fix this as just a normal member function..

Comment (p-durandin) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp:471]: LGTM, lets consider in next PR

Comment (moslex) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp:471]: Agreed with Vladimir P. that this comment should be addressed in the next PR. This PR can be merged to have green light to run nightly validation.

**Pull request merged**